packages/oauth/oauth_tests.js

```diff
@@ -59,3 +59,51 @@ Tinytest.add(
     test.equal(OAuth._retrievePendingCredential(key, secret), cred);
   }
 );
+
+Tinytest.add("oauth - _endOfLoginResponse with popup loginStyle supports ROOT_URL_PATH_PREFIX",
+  function (test) {
+    var rootUrlPathPrefix = __meteor_runtime_config__.ROOT_URL_PATH_PREFIX;
+    __meteor_runtime_config__.ROOT_URL_PATH_PREFIX = '/test-root-url-prefix';
+    var res = {
+      writeHead: function () {},
+      end: function (content) {
+        __meteor_runtime_config__.ROOT_URL_PATH_PREFIX = rootUrlPathPrefix;
+        test.matches(
+          content,
+          /\/test-root-url-prefix\/packages\/oauth\/end_of_popup_response\.js/
+        );
+      }
+    };
+    var details = {
+      credentials: {},
+      loginStyle: 'popup'
+    };
+    OAuth._endOfLoginResponse(res, details);
+  }
+);
+
+Tinytest.add("oauth - _endOfLoginResponse with redirect loginStyle supports ROOT_URL_PATH_PREFIX",
+  function (test) {
+    var rootUrlPathPrefix = __meteor_runtime_config__.ROOT_URL_PATH_PREFIX;
+    __meteor_runtime_config__.ROOT_URL_PATH_PREFIX = '/test-root-url-prefix';
+    var res = {
+      writeHead: function () {},
+      end: function (content) {
+        __meteor_runtime_config__.ROOT_URL_PATH_PREFIX = rootUrlPathPrefix;
+        test.matches(
+          content,
+          /\/test-root-url-prefix\/packages\/oauth\/end_of_redirect_response\.js/
+        );
+      }
+    };
+    var details = {
+      credentials: {},
+      loginStyle: 'redirect',
+      query: {
+        // {"redirectUrl": "http://localhost:3000/"}
```

Comment:
> The comment I added is there simply to make clear what's in the base64 encoded string.

Ah!  I didn't realize that it was the encoded version of `state`.  :smile:

Yes, I'd say change it to use the code you provided otherwise that message (`not on the same host as the app not on the same host as the app `) will print out in the (rare) cases when tests are run on a different port (for example, `./meteor --port 3600 test-packages oauth`).